llvm/lib/Transforms/Instrumentation/SPIRITTAnnotations.cpp

```diff
@@ -240,20 +246,24 @@ PreservedAnalyses SPIRITTAnnotationsPass::run(Module &M,
       SPIRV_GROUP_FMAX,      SPIRV_GROUP_UMAX, SPIRV_GROUP_SMAX};
 
   for (Function &F : M) {
-    // Annotate only SPIR kernels
-    if (F.isDeclaration() || !isSPIRKernel(F))
+    // Annotate only SPIR kernels and functions
+    bool IsSPIRKernel = isSPIRKernel(F);
+    bool IsSPIRFunction = isSPIRFunction(F);
+    if (F.isDeclaration() || !(isSPIRKernel || isSPIRFunction))
```

Comment:
```suggestion
    if (F.isDeclaration())
```
`(isSPIRKernel || isSPIRFunction)` is `true` for the module with SPIR target triple.

Reply:
Applied